### PR TITLE
`Assessment`: Fix the availability of the filter for locked submissions

### PR DESCRIPTION
**Pull request by Santia-go**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I implemented the changes with a good performance and prevented too many database calls.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.

### Motivation and Context

The issue [#5307](https://github.com/ls1intum/Artemis/issues/5307) presents the previous behavior. 
### Description

The filter for locked submissions is available just if there are already locked submissions. Otherwise no. 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Modeling/text/programming/upload file exercise 

1. Log in to Artemis (as instructor/tutor)
2. Navigate to Course Administration
3. Create 1 Modeling, 1 text Exercise, 1 programming exercise, and 1 upload file exercise ( all of them with the manual assessment).
{First case}
4. Check that without submissions, the filter for locked submissions **is not available** (_see screenshot 1_). 
{Second case}
4. As a student submit the solution for those exercises 
6. As an instructor, do not provide any assessment yet and don't lock any submission, and check the filter for locked submissions **is not available** (_see screenshot 2_).
{Third case}
7. Open some submissions and locked them. Then verify that the filter for locked submissions **is available** now (_see screenshot 3_). 
{Fourth case}
8. Unlock or complete the assessment for those submissions and check the filter for locked submissions **is not available** (_see screenshot 4_).

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|

### Screenshots

Screenshot 1
image

Screenshot 2
image

Screenshot 3
image

Screenshot 4
image

### CODE REVIEW

**Review by jschmautzer — APPROVED** [review]
tested on ts2, works as described :)
Is there a reason why the submissions of a text exercises are printed bold (like in screenshot 3) and the submissions for other exercises are printed normal (like the upload exercise in screenshot 4)?

File: src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.ts
Comment (Strohgelaender) [inline src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.ts:3]: unused import.
```suggestion
```

**Comment by Santia-go** [comment]
> tested on ts2, works as described :) Is there a reason why the submissions of a text exercises are printed bold (like in screenshot 3) and the submissions for other exercises are printed normal (like for the upload exercise in screenshot 4)?

That is a good question. I didn't notice that. 
I will take a look! 
Thank you!

**Review by jbllhff — DISMISSED** [review]
Tested on ts1, works fine!
Is it intended, that I can press "continue assessment" in the submissions board for programming exercises but only "open assessment" for every exercise else. In the former case, my feedback is cached, in the latter I have to re-enter the feedback.

**Comment by Santia-go** [comment]
> Tested on ts1, works fine! Is it intended, that I can press "continue assessment" in the submissions board for programming exercises but only "open assessment" for every exercise else. In the former case, my feedback is cached, in the latter I have to re-enter the feedback.

This is a good question. Actually Idk. 
But thanks for the observation. I will review why is in that way just in Programming exercises.

File: src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.ts
Comment (Hialus) [inline src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.ts:23]: You should consider renaming this. Currently the name suggests that this attribute contains the actual filtered submissions instead of only the amount of filtered submissions.

File: src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.ts
Comment (Hialus) [inline src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.ts:21]: Why did you put `@Input()` on a separate line here? You did not do the same with the `@Output()` below.
Please keep this consistent within a single file.

File: src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.ts
Comment (Hialus) [inline src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.ts:57]: Why aren't you using the `isSubmissionLocked` method here? It does the same as your code, so using it reduces code duplication.

File: src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.html
Comment (Hialus) [inline src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.html:1]: Is there any particular reason why you put the `*ngIf` inside of an `ng-container` instead of directly putting it into the `div`?
It should behave the same but avoids an unnecessary indentation.

File: src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.ts
Comment (Hialus) [reply src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.ts:57]: Also, does this have to be run on every change? This would also run this code whenever the filter changes (which would be redundant, as we already know that the result will be >1).
You should either move this into ngOnInit or first check if the submissions were changed as part of the OnChanges event. You can do this by adding and checking the `SimpleProps` to the `ngOnChanges` method: `ngOnChanges(changes: SimpleChanges)`

**Review by Hialus — CHANGES_REQUESTED** [review]
Reviewed the code and left some remarks.

**Review by jschmautzer — DISMISSED** [review]
re-approve - tested on ts2
(only this I still noticed: https://github.com/ls1intum/Artemis/pull/5308#issuecomment-1172042946)

**Review by nadinesamyy — APPROVED** [review]
tested on ts2, looks good

**Review by maximiliansoelch — DISMISSED** [review]
Code changes look good

**Review by mallgaier — APPROVED** [review]
Tested on TS1, works as expected 👍

**Review by rhubinger — APPROVED** [review]
Tested on TS2. Works as expected

**Review by luisa-we — DISMISSED** [review]
tested on ts1, works as expected

**Review by max-bergmann — APPROVED** [review]
Tested on TS2. Works as expected

**Review by Marvin-Vitcu — APPROVED** [review]
Tested on TS1. Works as intended.

**Review by MetinLamby — COMMENTED** [review]
for me, the filter for locked submission is available. even though the modeling exercise is assessed manually as you suggested
Screenshot 2022-07-10 at 11 55 09

**Review by yuling-sun — DISMISSED** [review]
Tested on TS1. Worked as described.

**Comment by SimonPaul02** [comment]
> for me, the filter for locked submission is available. even though the modeling exercise is assessed manually as you suggested Screenshot 2022-07-10 at 11 55 09

it worked for me. Did you deploy it correctly?

**Review by SimonPaul02 — APPROVED** [review]
Tested on TS1, worked as described

**Review by mallgaier — APPROVED** [review]
Re-approve after merge conflict

**Review by nadinesamyy — APPROVED** [review]
re-approve, code changes look good

**Review by maximiliansoelch — APPROVED** [review]
Reapprove

**Review by jschmautzer — APPROVED** [review]
re-approve, tested on ts2

I noticed something else, where I am not sure if this behavior is intended (only a small thing): In case of a programming / modeling / upload exercise, I click on "submissions" and then on "open assessment" / "assess submission". Then I always get the following path in Artemis (shown on the top left of the page): Course Management > [Name of Course] > Exercises > [Name of Exercise] > Submissions > Assessment.
However, if I follow the same steps for the text exercise I get the following path: Course Management > [Name of Course] > Exercises > [Name of Exercise] > **Participation** > Submissions > Assessment
I don't know if this is intended behavior or whats the reason for it - however, it is only a small bug, which does not correlate to this PR.

**Pull request merged**
